src/Ship.java

```diff
@@ -12,7 +12,7 @@ public Ship(int life) {
 	// in the observer pattern
 	//
 	public void update() throws Exception {
-
+		hit();
```

Comment:
Notice that this project utilizes the `this` keyword in contexts where it's not always needed. It's best to be consistent throughout the entire codebase.